### PR TITLE
Fix error on empty draggable context

### PR DESCRIPTION
**Pull request by Andries-Smit**
- When the there was no context for the draggable widget, an error was throwing. Now the widget will be disabled. (Context is sometimes emptied by the platform when closing an form)
 - Remove subscriptions: no data is needed from the context object except for the GUID and type, which will not change even on updates from "subscriptions"
 - Remove unused code and formatting
 - Update version widget version number and add a test case in the Mendix test project for the empty context

### CODE REVIEW

File: src/DragAndDropSupportWidget/widget/DroppableWidget.js [hunk 1]
```diff
@@ -1,117 +1,67 @@
...
-                node,
-                thisObj = this;
-            
-            if (this.acceptSelector) {
-                args.accept = this.acceptSelector;
+            var self = this;
+            // http://api.jqueryui.com/droppable/
+            var args = {
+                accept: this.acceptSelector || "*",
+                reedy: true,
```
Comment (mgroeneweg) [inline src/DragAndDropSupportWidget/widget/DroppableWidget.js]: Are you sure this is correct? reedy i.s.o. greedy?

File: src/DragAndDropSupportWidget/widget/DroppableWidget.js [hunk 2]
```diff
@@ -1,117 +1,67 @@
...
             } else {
                 node = $(this.domNode.parentElement);
             }
             node.addClass(this.dropTargetClass).droppable(args);
-         
         },
-        
+
         _drop: function (event, ui) {
-            var droppedObjectType,
```
Comment (mgroeneweg) [inline src/DragAndDropSupportWidget/widget/DroppableWidget.js]: As per convention, variables are to be defined at the top of the function rather than inline. Why did you change this?

File: src/DragAndDropSupportWidget/widget/DroppableWidget.js [hunk 3]
```diff
@@ -122,25 +72,19 @@ require( [
...
-
         /**
          * Called upon creation of onDropEntity
          *
          * @param mendixObject  The new Mendix object
          */
         onDropObjectCreated : function (mendixObject) {
-
             // console.log("onDropObjectCreated");
-            var draggedReference,
```
Comment (mgroeneweg) [inline src/DragAndDropSupportWidget/widget/DroppableWidget.js]: As per convention, variables are to be defined at the top of the function rather than inline. Why did you change this?

File: src/DragAndDropSupportWidget/widget/DroppableWidget.js [hunk 4]
```diff
@@ -1,117 +1,67 @@
...
-
-        },
-
-        // mxui.widget._WidgetBase.uninitialize is called when the widget is destroyed. Implement to do special tear-down work.
-        uninitialize: function () {
-            // Clean up listeners, helper objects, etc. There is no need to remove listeners added with this.connect / this.subscribe / this.own.
-        },
-
         _setupEvents: function () {
-            var args = {},
```
Comment (mgroeneweg) [inline src/DragAndDropSupportWidget/widget/DroppableWidget.js]: As per convention, variables are to be defined at the top of the function rather than inline. Why did you change this?

**Comment by mgroeneweg** [comment]
First of all, I really appreciate the effort you put in by bringing the code forward. Please have a look at the individual comments.

File: src/DragAndDropSupportWidget/widget/DroppableWidget.js (hunk 1 above)
Comment (Andries-Smit) [reply src/DragAndDropSupportWidget/widget/DroppableWidget.js]: Yes, see docs, http://api.jqueryui.com/droppable/#option-accept default `*`

File: src/DragAndDropSupportWidget/widget/DroppableWidget.js (hunk 2 above)
Comment (Andries-Smit) [reply src/DragAndDropSupportWidget/widget/DroppableWidget.js]: Good question. Good code is written for readability, low maintainable and preventing bugs. This convention is correct preventing accidentally creating variables that might out of scope later on. Though modern IDE will warn you when making such mistakes. Especially when you code "strict" change are less. Noising moving the stuff around though keeping the order makes it less mind boggling. 
https://www.w3schools.com/js/js_hoisting.asp

In this case the less code makes it easier to read.

File: src/DragAndDropSupportWidget/widget/DroppableWidget.js (hunk 3 above)
Comment (Andries-Smit) [reply src/DragAndDropSupportWidget/widget/DroppableWidget.js]: see above

File: src/DragAndDropSupportWidget/widget/DroppableWidget.js (hunk 4 above)
Comment (Andries-Smit) [reply src/DragAndDropSupportWidget/widget/DroppableWidget.js]: see above

**Comment by Andries-Smit** [comment]
Thanks, good to see good questions...

File: src/DragAndDropSupportWidget/widget/DroppableWidget.js (hunk 1 above)
Comment (mgroeneweg) [reply src/DragAndDropSupportWidget/widget/DroppableWidget.js]: I suppose the *reedy* is just a typo then. Almost leaving for the airport. This will have to wait to mid August.

**Comment by mgroeneweg** [comment]
Had some trouble merging it but it went through all right. Thanks for all the effort. I did change the reedy into greedy though.
